preference/src/main/kotlin/store/PreferenceStore.kt

```diff
@@ -1,14 +1,16 @@
 package org.jellyfin.preference.store
 
 import org.jellyfin.preference.Preference
+import org.jellyfin.preference.migration.MigrationContext
 
 /**
  * Abstract class defining the required functions for a preference store.
  *
  * This implements shared functionality (such as Enum handling), whilst allowing for
  * different backing stores.
  */
-abstract class PreferenceStore {
+@Suppress("TooManyFunctions")
+abstract class PreferenceStore<ME, MV> {
```

Comment:
When <ME, MV> is used in this PR, it represents `Migration<E, V>`, right?

```

data class Migration<E, V>(
		val toVersion: Int,
		val body: E.(V) -> Unit
	)

```

I'm not sure what this is doing, specifically the `E.(V) -> Unit` part. What would I google to read up on it?
I'm already reading about data classes

Reply:
> When <ME, MV> is used in this PR, it represents Migration<E, V>, right?

Yes, the "M" prefix is used to indicate it's for the migration part.

The "ME" means the editor type to use and the "MV" means the current data to send to the editor. You can look at the SharedPreferencesStore to see how it's used:

`PreferenceStore<SharedPreferences.Editor, SharedPreferences>()`

Which then causes the "runMigrations" function to use SharedPreferences.Editor as this scope and SharedPreferences as "it". (See usage in UserPreferences).

> I'm not sure what this is doing, specifically the E.(V) -> Unit part.

`() -> Unit` would mean: a function with no parameters that returns nothing (Unit)
`(V) -> Unit` would mean that is contains one parameter, of type V
`E.(V) -> Unit` would mean that the function is in the (this) scope of type E

Reply:
That makes sense now, thanks for the explanation 🙂